pynxtools/dataconverter/convert.py

```diff
@@ -348,29 +356,44 @@ def convert_cli(
     if params_file:
         try:
             convert(**parse_params_file(params_file))
+            return
         except TypeError as exc:
             sys.tracebacklimit = 0
-            raise TypeError(
+            raise click.UsageError(
                 (
                     "Please make sure you have the following entries in your "
                     "parameter file:\n\n# NeXusParser Parameter File - v0.0.1"
                     "\n\ndataconverter:\n\treader: value\n\tnxdl: value\n\tin"
                     "put-file: value"
                 )
             ) from exc
-    else:
-        if nxdl is None:
-            sys.tracebacklimit = 0
-            raise IOError(
-                "\nError: Please supply an NXDL file with the option:"
-                " --nxdl <path to NXDL>"
-            )
-        if mapping:
-            reader = "json_map"
-            if mapping:
-                input_file = input_file + tuple([mapping])
-        convert(input_file, reader, nxdl, output, generate_template, fair, undocumented)
+    if nxdl is None:
+        raise click.UsageError("Missing option '--nxdl'")
+    if mapping:
+        reader = "json_map"
+        input_file = input_file + tuple([mapping])
```

Comment:
I think this should be moved back in the else statement. This will also not work with the --params if an --nxdl is not provided, right?

Reply:
It will be fine, I added a `return` statement in the `try` block for the params config to get rid of the additional else level.

Reply:
But maybe let's add a testing case for the params file before we merge this. To not break it again

Reply:
You're right. I just saw it a second ago. Thank you. I'll mark this as resolved.

Reply:
I added the test @lukaspie @sherjeelshabih

Reply:
Great, feel free to merge then 👍